### PR TITLE
bigquery: document new methods

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

**Comment by pongad** [comment]
@garrettjonesgoogle Can you teach me to run snippet integration tests? Running `mvn verify` on the example dir ran no tests.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java [hunk 1]
```diff
@@ -465,6 +465,43 @@ public Table createTable(String datasetName, String tableName, String fieldName)
...
+  // [TARGET listTableData(String, String, Schema, TableDataListOption...)]
+  // [VARIABLE "my_dataset_name"]
+  // [VARIABLE "my_table_name"]
+  // [VARIABLE ...]
+  // [VARIABLE "field"]
+  public Page<FieldValueList> listTableDataSchema(
+      String datasetName, String tableName, Schema schema, String field) {
+    // [START listTableDataSchema]
+    Page<FieldValueList> tableData =
+        bigquery.listTableData(datasetName, tableName, schema, TableDataListOption.pageSize(100));
```
Comment (tswast) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: I'm moderately worried someone might see the page size option and think they need to set that and/or that 100 is a recommended page size.

Maybe worth adding a comment clarifying that the options are optional?

Comment (pongad) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Done

Comment (tswast) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/BigQuerySnippets.java]: Thanks.

**Pull request merged**